wordpress.org/public_html/wp-content/plugins/official-wordpress-events/template-events.php

```diff
@@ -9,20 +9,26 @@
 		<ul class="ofe-event-list">
```

Comment:
I'm guessing we should also apply these changes to `template-events-online.php` (which generates https://make.wordpress.org/community/events/online/)

Reply:
I'm wondering is this page still in use? I didn't seem to able to find a link redirecting to it, and also found that in `/events`, online events were already included. Perhaps there's a post/thread discussing about this, I'll see if I can find it out.

Do you think if it's the case, we can just remove the related files? Or it's the best to apply the changes here to the file mentioned first and keep the codes for future need.

Reply:
Updated in 840d20b61a2669d86b864e09559e24148016bde3

![image](https://github.com/WordPress/wordpress.org/assets/18050944/bc9ca9fd-4310-4ca1-bfc9-98ebf73db6a4)

Reply:
`Type` hasn't been added yet as I've found that adding it doesn't seem to provide much help currently, as shown below, it's all `meetups`. However, there have been instances of online WordCamp in the past few years, and from what I understand, upcoming `pilot events` are also very likely to be held online? If that's the case, then the `type` should also be added.

![meetup](https://github.com/WordPress/wordpress.org/assets/18050944/c8e7847d-60dd-4ab6-9d3f-473f0c1a8891)

Reply:
Note: While modifying the `events/online` page, I ran into the following issue which is the best to be resolved in another PR. I'll document it here first. There are some pictures and contexts that can aid in the explanation. If there are any thoughts, feel free to comment. If not, I'll open a ticket in Trac later for future fix.

There seems to be some discrepancy between the time displayed on the `events/online` page and the `events` page. On the `events`, it displays the local time of the event's location, sorted from earliest to latest. 

But on the `events/online`, the time is converted to the user's local time, with the timezone added afterward, such as UTC+8 that I am seeing. And in this case, it is no longer sorted from earliest to latest.

One possible simple improvement I can think of is to always display the time in the event's local time, along with that time zone. This could also retain the sort order from earliest to latest.

Reply:
> If that's the case, then the type should also be added.

Yeah, I think so. There could be some WordCamps in the future that choose to do an online event too, so IMO it's best if the template can handle it.

> On the events, it displays the local time of the event's location... But on the events/online, the time is converted to the user's local time

IIRC that was intentional, since the user's time is what matters most to online viewers. For in-person events, attendees would need to know the venue time. The sorting might be a bug though.

@ryelle , does that seem right to you?

Reply:
Updated in 8e8f067f817d095d41460228ca8fd54a4c8ab42f.

Case conversion can also be done [after this line](https://github.com/WordPress/wordpress.org/blob/trunk/wordpress.org/public_html/wp-content/plugins/official-wordpress-events/official-events-online.php#L45) with something like `'type' => 'wordcamp' === $event->type ? 'WordCamp' : mb_convert_case( $event->type, MB_CASE_TITLE, "UTF-8" )`, but I assume it's likely that there'd be a different idea about how to use the value elsewhere someday.

Reply:
> IIRC that was intentional, since the user's time is what matters most to online viewers. For in-person events, attendees would need to know the venue time.

Correct about the timezone.

> The sorting might be a bug though.

Yeah… the sorting is a bug… `getSortedEvents` assumes the original list is presorted, because it seems like it is— but it's sorted by the `date_utc` value in the database, which as [this comment notes,](https://github.com/WordPress/wordpress.org/blob/376bdd6e4866a8600a757d89496c0fc5bc97f096/wordpress.org/public_html/wp-content/plugins/official-wordpress-events/official-events-online.php#L39-L40) is actually not UTC, but the local timezone. So something at 5pm in London (16 UTC) would appear after something at 4pm in New York (20 UTC), despite the London event happening first.

Actually sorting the events in `getSortedEvents` should fix that.

Reply:
Trac ticket: https://meta.trac.wordpress.org/ticket/7071
PR opened: https://github.com/WordPress/wordpress.org/pull/158